Sources/SwiftBSON/BSONDocument.swift

```diff
@@ -517,3 +517,42 @@ extension BSONDocument: BSONValue {
 extension BSONDocument: CustomStringConvertible {
     public var description: String { self.toExtendedJSONString() }
 }
+
+extension BSONDocument {
+    /**
+     * Returns whether this BSONDocument contains exactly the same key/value pairs as the provided BSONDocument,
+     * regardless of the order of the keys.
+     *
+     * Note: This method is much less efficient than checking for regular equality since the document is internally
+     * ordered.
+     *
+     * - Parameters:
+     *   - other: a `BSONDocument` to compare this document with.
+     *
+     * - Returns: a `Bool` indicating whether the two documents are equal.
+     */
+    public func equalsIgnoreKeyOrder(_ other: BSONDocument) -> Bool {
+        let a = self
+        let b = other
+        let keysA = a.keys.sorted()
+        let keysB = b.keys.sorted()
+
+        guard keysA.count == keysB.count else {
+            return false
+        }
+
+        for (k, v) in self {
+            if case let (.document(aDoc), .document(bDoc)?) = (v, other[k]) {
+                guard aDoc.equalsIgnoreKeyOrder(bDoc) else {
+                    return false
+                }
+                continue
+            }
+            guard v == b[k] else {
```

Comment:
we seem to be mixing usage of `other` and `b` here. we can probably get rid of `a` and `b` above and just use `self` and `other` directly